### PR TITLE
[AIRFLOW-5597] Linkify urls in task instance log

### PR DESCRIPTION
**Pull request by ZxMYS**
#6070 # Jira

- [x] My PR addresses the following [Airflow Jira]
  - https://issues.apache.org/jira/browse/AIRFLOW-5597

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Very often we have urls in our task instances' logs. This change allows these urls to be rendered as clickable links, so users no longer have to copy and paste these urls manually.
image

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Just a simple JS change

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.

### CODE REVIEW

File: airflow/www/templates/airflow/ti_log.html [hunk 1]
```diff
@@ -121,10 +121,18 @@ <h4>{{ title }}</h4>
...
             if(auto_tailing && checkAutoTailingCondition()) {
-              var should_scroll = true
+              var should_scroll = true;
             }
             // The message may contain HTML, so either have to escape it or write it as text.
-            document.getElementById(`try-${try_number}`).textContent += res.message + "\n";
+            var escaped_message = $("<div/>").text(res.message).html();
+
+            // Detect urls
+            var url_regex = /(https?:\/\/[^\s<>"]+)/g;
```
Comment (kaxil) [inline airflow/www/templates/airflow/ti_log.html]: Will this cover all cases?

How about the following from http://www.noah.org/wiki/RegEx_Python?

```
http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+
```

File: airflow/www/templates/airflow/ti_log.html [hunk 2]
```diff
@@ -121,10 +121,18 @@ <h4>{{ title }}</h4>
           if (res.message) {
             // Auto scroll window to the end if current window location is near the end.
             if(auto_tailing && checkAutoTailingCondition()) {
-              var should_scroll = true
+              var should_scroll = true;
             }
             // The message may contain HTML, so either have to escape it or write it as text.
-            document.getElementById(`try-${try_number}`).textContent += res.message + "\n";
+            var escaped_message = $("<div/>").text(res.message).html();
```
Comment (mik-laj) [inline airflow/www/templates/airflow/ti_log.html]: Can you use a non-jquery function? We have [escpeHtml](https://github.com/ZxMYS/incubator-airflow/blob/xzhu/linkify/airflow/www/static/js/base.js#L34-L37) function

File: airflow/www/templates/airflow/ti_log.html (hunk 1 above)
Comment (ZxMYS) [reply airflow/www/templates/airflow/ti_log.html]: This regex doesn't work when there's a anchor tag (#). I switched to another more restrict regex - https://www.regexpal.com/?fam=111808

**Comment by ZxMYS** [comment]
Is there a way to re-run the checks? I don't think the error in travis ci is related to my change.

**Comment by mik-laj** [comment]
I restarted this Travis's job.

**Comment by aoen** [comment]
LGTM

**Comment by ZxMYS** [comment]
To future readers: feel free to improve the regex. The regex in this commit may have been too strict and limiting (e.g. only checks for http / https links, and not allowing Internationalized Domain Names, etc.)

**Pull request merged**